File: q80-q89/q82/q82.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteDuplicates(self, head: ListNode) -> ListNode:
        isDuplicate = False
        if head == None:
            return head;
        while head.next and head.val == head.next.val:
            head = head.next
            isDuplicate = True
        if isDuplicate:
            head = self.deleteDuplicates(head.next)
        else:
            head.next = self.deleteDuplicates(head.next)
        return head;

File: q1-q99/q80-q89/q82/q82.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        if (!head) 
            return head;
        bool dup_exist = false; 
        while (head->next && head->val == head->next->val) {
            head = head->next;
            dup_exist = true;
        }
        if (dup_exist) 
            head = deleteDuplicates(head->next);
        else 
            head->next = deleteDuplicates(head->next);
        return head;
    }
};
